app/scripts/metamask-controller.js

```diff
@@ -2351,14 +2350,19 @@ export default class MetamaskController extends EventEmitter {
       // state has account balance before it is synced with UI
       await this.accountTracker._updateAccounts();
     } finally {
-      // Message startUISync is used in MV3 to start syncing state with UI
-      // Sending this message after login is completed helps to ensure that incomplete state without
-      // account details are not flushed to UI.
+      this._startUISync();
+    }
+  }
+
+  _startUISync() {
+    // Message startUISync is used in MV3 to start syncing state with UI
+    // Sending this message after login is completed helps to ensure that incomplete state without
+    // account details are not flushed to UI.
+    if (this.activeControllerConnections > 0) {
       this.emit('startUISync');
-      // During controllers initialisation memstore updates a lot, delaying this subscription
-      // helps to delay early flusing of state to UI.
-      this.memStore.subscribe(this.sendUpdate.bind(this));
     }
+    this.startUISync = true;
+    this.memStore.subscribe(this.sendUpdate.bind(this));
   }
```

Comment:
The change is also useful in scenario when service worker restart before user started onboarding